tests/zkevm/test_worst_bytecode.py

```diff
@@ -1,59 +1,172 @@
+            size=Op.EXTCODESIZE(initcode_address),
+        )
+        + Op.MSTORE(
+            0,
+            Op.CREATE(
+                value=0,
+                offset=0,
+                size=Op.MSIZE,
+            ),
+        )
+        + Op.RETURN(0, 32)
+    )
+    factory_address = pre.deploy_contract(code=factory_code)
+
+    # The factory caller will call the factory contract N times, creating N new contracts.
+    # Calldata should contain the N value.
+    factory_caller_code = Op.CALLDATALOAD(0) + While(
+        body=Op.POP(Op.CALL(address=factory_address)),
+        condition=Op.PUSH1(1) + Op.SWAP1 + Op.SUB + Op.DUP1 + Op.ISZERO + Op.ISZERO,
+    )
+    factory_caller_address = pre.deploy_contract(code=factory_caller_code)
+
+    # TODO: The correct way to calculate this is to use the gas costs and the transaction
+    # intrinsic cost calculator, but this is generating approx 13,822 calls, and deploying this
+    # many different contracts requires 1,974 blocks.
+    # gas_costs = fork.gas_costs()
+    # intrinsic_gas_cost_calc = fork.transaction_intrinsic_cost_calculator()
+    # max_number_of_contract_calls = (GAS_LIMIT - intrinsic_gas_cost_calc()) // (
+    #     gas_costs.G_VERY_LOW + gas_costs.G_BASE + gas_costs.G_COLD_ACCOUNT_ACCESS
+    # )
+    max_number_of_contract_calls = 10
+    total_contracts_to_deploy = max_number_of_contract_calls
+    approximate_gas_per_deployment = 4_970_000  # Obtained from evm tracing
+    contracts_deployed_per_tx = GAS_LIMIT // approximate_gas_per_deployment
+
+    deploy_txs = []
+
+    def generate_deploy_tx(contracts_to_deploy: int):
+        return Transaction(
+            to=factory_caller_address,
+            gas_limit=GAS_LIMIT,
+            gas_price=10**9,  # Bump required due to the amount of full blocks
+            data=Hash(contracts_deployed_per_tx),
+            sender=pre.fund_eoa(),
+        )
+
+    for _ in range(total_contracts_to_deploy // contracts_deployed_per_tx):
+        deploy_txs.append(generate_deploy_tx(contracts_deployed_per_tx))
+
+    if total_contracts_to_deploy % contracts_deployed_per_tx != 0:
+        deploy_txs.append(
+            generate_deploy_tx(total_contracts_to_deploy % contracts_deployed_per_tx)
+        )
 
-    attack_code = sum([Op.EXTCODESIZE(contract_addrs[i]) for i in range(num_called_contracts)])
-    attack_contract = pre.deploy_contract(code=attack_code)
+    post = {}
+    deployed_contract_addresses = []
+    for i in range(total_contracts_to_deploy):
+        deployed_contract_address = compute_create_address(
+            address=factory_address,
+            nonce=i + 1,
+        )
+        post[deployed_contract_address] = Account(nonce=1)
+        deployed_contract_addresses.append(deployed_contract_address)
 
-    tx = Transaction(
-        to=attack_contract,
+    opcode_code = sum(opcode(address=address) for address in deployed_contract_addresses) + Op.STOP
+    opcode_address = pre.deploy_contract(code=opcode_code)
```

Comment:
I think we need a TODO here (and potentially add an assertion).

For the actual worst case, the would be 13k calls, but that `EXTCODESIZE(address)+POP` is 20+1+1 bytes, so that means takes 13k*22~=286KiB which is bigger than the contract size.

As a sanity check, I would add:
```
if len(opcode_code) > MAX_CODE_SIZE:
        raise ValueError(f"Code size {len(code)} exceeds maximum code size {MAX_CODE_SIZE}")
```
No?

And a TODO to think of a way of working around this limitation. Feels to me we might need to switch to storing the addresses in storage slots and doing some SLOAD+EXTCODESIZE.